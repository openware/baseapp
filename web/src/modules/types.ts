import { MarketsState } from './public/markets';
import { OrderBookState } from './public/orderBook';
import { OrdersState } from './user/orders';

export interface CommonError {
    code: number;
    message: string[];
}

export interface CommonState {
    error?: CommonError;
    loading?: boolean;
}

export type OrderStatus = 'wait' | 'done' | 'cancel' | 'pending' | 'reject' | 'trigger_wait';
export type OrderSide = 'sell' | 'buy';
export type OrderType = 'limit' | 'market' | 'stop_loss' | 'stop_limit' | 'take_profit' | 'take_limit';
export type OrderKind = 'bid' | 'ask';

export interface OrderCommon {
    price: string;
    state: OrderStatus;
    remaining_volume: string;
    origin_volume: string;
    executed_volume?: string;
    side: OrderSide;
    market: string;
    ord_type?: OrderType;
    avg_price?: string;
    volume?: number;
    trigger_price?: string;
    created_at?: string;
    updated_at?: string;
    triggered_at?: string;
    confirmed?: boolean;
    uuid?: string;
    id?: number;
    kind?: OrderKind;
    trades_count?: number;
}

export interface OrderEvent extends OrderCommon {
    at: number;
    order_type?: OrderType;
}

export interface MarketUpdateEvent {
    asks: Array<[number, number]>;
    bids: Array<[number, number]>;
}

export interface CoreState {
    orders: OrdersState;
    orderBook: OrderBookState;
    markets: MarketsState;
}

export interface ClientState {
    coreData: CoreState;
}
