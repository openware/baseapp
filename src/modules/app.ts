import { combineReducers } from 'redux';
import { alertReducer  } from './public/alert';
import { changeColorThemeReducer  } from './public/colorTheme';
import { configsReducer } from './public/configs';
import { currenciesReducer } from './public/currencies';
import { gridLayoutReducer } from './public/gridLayout/reducer';
import { changeLanguageReducer  } from './public/i18n';
import { klineReducer  } from './public/kline';
import { marketsReducer } from './public/markets';
import { memberLevelsReducer } from './public/memberLevels';
import {
    depthReducer,
    incrementDepthReducer,
    orderBookReducer,
} from './public/orderBook';
import { rangerReducer  } from './public/ranger/reducer';
import { recentTradesReducer  } from './public/recentTrades';
import { apiKeysReducer } from './user/apiKeys';
import { authReducer  } from './user/auth';
import { beneficiariesReducer } from './user/beneficiaries';
import { getGeetestCaptchaReducer } from './user/captcha';
import { sendEmailVerificationReducer } from './user/emailVerification';
import { historyReducer  } from './user/history';
import {
    documentsReducer,
    identityReducer,
    labelReducer,
    phoneReducer,
} from './user/kyc';
import { newHistoryReducer } from './user/newHistory';
import { openOrdersReducer } from './user/openOrders';
import { ordersReducer  } from './user/orders';
import { ordersHistoryReducer  } from './user/ordersHistory';
import { passwordReducer  } from './user/password';
import { profileReducer  } from './user/profile';
import { userActivityReducer  } from './user/userActivity';
import { walletsReducer  } from './user/wallets';
import { withdrawLimitReducer  } from './user/withdrawLimit';

export const publicReducer = combineReducers({
    colorTheme: changeColorThemeReducer,
    configs: configsReducer,
    currencies: currenciesReducer,
    recentTrades: recentTradesReducer,
    markets: marketsReducer,
    orderBook: orderBookReducer,
    depth: depthReducer,
    incrementDepth: incrementDepthReducer,
    ranger: rangerReducer,
    i18n: changeLanguageReducer,
    kline: klineReducer,
    alerts: alertReducer,
    rgl: gridLayoutReducer,
    memberLevels: memberLevelsReducer,
});

export const userReducer = combineReducers({
    auth: authReducer,
    beneficiaries: beneficiariesReducer,
    label: labelReducer,
    orders: ordersReducer,
    password: passwordReducer,
    profile: profileReducer,
    wallets: walletsReducer,
    phone: phoneReducer,
    identity: identityReducer,
    documents: documentsReducer,
    history: historyReducer,
    newHistory: newHistoryReducer,
    apiKeys: apiKeysReducer,
    userActivity: userActivityReducer,
    ordersHistory: ordersHistoryReducer,
    openOrders: openOrdersReducer,
    sendEmailVerification: sendEmailVerificationReducer,
    captchaKeys: getGeetestCaptchaReducer,
    withdrawLimit: withdrawLimitReducer,
});
