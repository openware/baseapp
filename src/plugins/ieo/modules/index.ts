export * from './public';
export * from './user';
export * from './app';
