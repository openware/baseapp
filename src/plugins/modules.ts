import { combineReducers } from 'redux';
// tslint:disable-next-line no-submodule-imports
import { all, call } from 'redux-saga/effects';
import {
    ieoReducer,
    rootIEOSaga,
    StateIEO,
} from './ieo';

export interface PluginsState {
    ieo: StateIEO;
}

export const pluginsReducer = combineReducers({
    ieo: ieoReducer,
});

export function* rootPluginsSaga() {
    yield all([
        call(rootIEOSaga),
    ]);
}
