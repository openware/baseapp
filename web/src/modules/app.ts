import { combineReducers } from 'redux';
import { configUpdateReducer } from './admin/config';
import { alertReducer  } from './public/alert';
import { blockchainsReducer } from './public/blockchains';
import { blocklistAccessReducer } from './public/blocklistAccess';
import { configsReducer } from './public/configs';
import { currenciesReducer } from './public/currencies';
import { errorHandlerReducer } from './public/errorHandler';
import { changeColorThemeReducer  } from './public/globalSettings';
import { gridLayoutReducer } from './public/gridLayout';
import { changeLanguageReducer  } from './public/i18n';
import { klineReducer  } from './public/kline';
import { marketsReducer } from './public/markets';
import { memberLevelsReducer } from './public/memberLevels';
import { depthReducer, incrementDepthReducer, orderBookReducer } from './public/orderBook';
import { rangerReducer  } from './public/ranger/reducer';
import { recentTradesReducer  } from './public/recentTrades';
import { withdrawLimitsReducer  } from './public/withdrawLimits';
import { apiKeysReducer } from './user/apiKeys';
import { abilitiesReducer } from './user/abilities';
import { authReducer  } from './user/auth';
import { beneficiariesReducer } from './user/beneficiaries';
import { getGeetestCaptchaReducer } from './user/captcha';
import { documentationReducer } from './user/documentation';
import { sendEmailVerificationReducer } from './user/emailVerification';
import { historyReducer  } from './user/history';
import { internalTransfersReducer } from './user/internalTransfers';
import {
    addressesReducer,
    documentsReducer,
    identityReducer,
    labelReducer,
    phoneReducer,
} from './user/kyc';
import { openOrdersReducer } from './user/openOrders';
import { ordersReducer  } from './user/orders';
import { ordersHistoryReducer  } from './user/ordersHistory';
import { passwordReducer  } from './user/password';
import { profileReducer  } from './user/profile';
import { userActivityReducer  } from './user/userActivity';
import { walletsReducer  } from './user/wallets';
import { withdrawLimitReducer  } from './user/withdrawLimit';
import { marketsAdminReducer } from './admin/markets';
import { platformCreateReducer } from './admin/platform';
import { quickExchangeReducer } from './user/quickExchange';
import { p2pReducer } from './public/p2p';
import { paymentMethodReducer } from './user/paymentMethod';
import { p2pOffersReducer } from './user/p2pOffers';
import { p2pTransfersReducer } from './user/p2pTransfers';
import { p2pOrdersReducer } from './user/p2pOrders';
import { p2pDisputeReducer } from './user/p2pDispute';
import { feeGroupReducer } from './user/feeGroup';

export const publicReducer = combineReducers({
    alerts: alertReducer,
    blockchains: blockchainsReducer,
    blocklistAccess: blocklistAccessReducer,
    colorTheme: changeColorThemeReducer,
    configs: configsReducer,
    currencies: currenciesReducer,
    errorHandler: errorHandlerReducer,
    rgl: gridLayoutReducer,
    i18n: changeLanguageReducer,
    kline: klineReducer,
    markets: marketsReducer,
    memberLevels: memberLevelsReducer,
    orderBook: orderBookReducer,
    depth: depthReducer,
    incrementDepth: incrementDepthReducer,
    ranger: rangerReducer,
    recentTrades: recentTradesReducer,
    p2p: p2pReducer,
    withdrawLimits: withdrawLimitsReducer,
});

export const userReducer = combineReducers({
    addresses: addressesReducer,
    apiKeys: apiKeysReducer,
    auth: authReducer,
    beneficiaries: beneficiariesReducer,
    captcha: getGeetestCaptchaReducer,
    documentation: documentationReducer,
    history: historyReducer,
    documents: documentsReducer,
    identity: identityReducer,
    label: labelReducer,
    phone: phoneReducer,
    openOrders: openOrdersReducer,
    orders: ordersReducer,
    ordersHistory: ordersHistoryReducer,
    password: passwordReducer,
    profile: profileReducer,
    sendEmailVerification: sendEmailVerificationReducer,
    userActivity: userActivityReducer,
    wallets: walletsReducer,
    feeGroup: feeGroupReducer,
    withdrawLimit: withdrawLimitReducer,
    internalTransfers: internalTransfersReducer,
    quickExchange: quickExchangeReducer,
    abilities: abilitiesReducer,
    paymentMethod: paymentMethodReducer,
    p2pOffers: p2pOffersReducer,
    p2pTransfers: p2pTransfersReducer,
    p2pOrders: p2pOrdersReducer,
    p2pDispute: p2pDisputeReducer,
});

export const adminReducer = combineReducers({
    configUpdate: configUpdateReducer,
    markets: marketsAdminReducer,
    platform: platformCreateReducer,
});
