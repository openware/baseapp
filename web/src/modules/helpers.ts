import { MakerType } from './user/history';

const makerTypeMap = {
    ask: 'sell',
    bid: 'buy',
};

export const kindToMakerType = (kind: string): MakerType => makerTypeMap[kind];
