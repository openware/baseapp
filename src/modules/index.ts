import { combineReducers } from 'redux';
// tslint:disable-next-line no-submodule-imports
import { all, call } from 'redux-saga/effects';
import { publicReducer, userReducer } from './app';
import { AlertState, rootHandleAlertSaga } from './public/alert';
import { ColorThemeState } from './public/colorTheme';
import { ConfigsState, rootConfigsSaga } from './public/configs';
import { CurrenciesState, rootCurrenciesSaga } from './public/currencies';
import { GridLayoutState } from './public/gridLayout';
import { LanguageState } from './public/i18n';
import { KlineState, rootKlineFetchSaga } from './public/kline';
import { MarketsState, rootMarketsSaga } from './public/markets';
import {
    DepthIncrementState,
    DepthState,
    OrderBookState,
    rootOrderBookSaga,
} from './public/orderBook';
import { RangerState } from './public/ranger/reducer';
import { RecentTradesState, rootRecentTradesSaga } from './public/recentTrades';
import { ApiKeysState } from './user/apiKeys';
import { rootApiKeysSaga } from './user/apiKeys/sagas';
import { AuthState, rootAuthSaga } from './user/auth';
import { BeneficiariesState, rootBeneficiariesSaga } from './user/beneficiaries';
import { GeetestCaptchaState, rootGeetestCaptchaSaga } from './user/captcha';
import { EmailVerificationState, rootEmailVerificationSaga } from './user/emailVerification';
import { HistoryState, rootHistorySaga } from './user/history';
import { DocumentsState, rootSendDocumentsSaga } from './user/kyc/documents';
import { IdentityState, rootSendIdentitySaga } from './user/kyc/identity';
import { LabelState, rootLabelSaga } from './user/kyc/label';
import { PhoneState, rootSendCodeSaga } from './user/kyc/phone';
import { NewHistoryState, rootNewHistorySaga } from './user/newHistory';
import { OpenOrdersState, rootOpenOrdersSaga } from './user/openOrders';
import { OrdersState, rootOrdersSaga } from './user/orders';
import { OrdersHistoryState, rootOrdersHistorySaga } from './user/ordersHistory';
import { PasswordState, rootPasswordSaga } from './user/password';
import { ProfileState, rootProfileSaga } from './user/profile';
import { rootUserActivitySaga, UserActivityState } from './user/userActivity';
import { rootWalletsSaga, WalletsState } from './user/wallets';
import { rootWithdrawLimitSaga, WithdrawLimitState } from './user/withdrawLimit';
import { MemberLevelsState, rootMemberLevelsSaga } from './public/memberLevels';

export * from './public/markets';
export * from './public/orderBook';
export * from './public/colorTheme';
export * from './public/configs';
export * from './public/currencies';
export * from './public/i18n';
export * from './public/kline';
export * from './public/alert';
export * from './user/apiKeys';
export * from './user/auth';
export * from './user/beneficiaries';
export * from './user/captcha';
export * from './user/wallets';
export * from './user/profile';
export * from './user/openOrders';
export * from './user/orders';
export * from './user/ordersHistory';
export * from './user/password';
export * from './user/userActivity';
export * from './user/history';
export * from './user/newHistory';
export * from './user/kyc';
export * from './user/emailVerification';
export * from './user/withdrawLimit';
export * from './public/memberLevels';

export interface RootState {
    public: {
        colorTheme: ColorThemeState;
        configs: ConfigsState;
        currencies: CurrenciesState;
        recentTrades: RecentTradesState;
        markets: MarketsState;
        orderBook: OrderBookState;
        depth: DepthState;
        incrementDepth: DepthIncrementState;
        ranger: RangerState;
        i18n: LanguageState;
        alerts: AlertState;
        kline: KlineState;
        rgl: GridLayoutState;
        memberLevels: MemberLevelsState;
    };
    user: {
        auth: AuthState;
        beneficiaries: BeneficiariesState;
        orders: OrdersState;
        password: PasswordState;
        profile: ProfileState;
        label: LabelState;
        wallets: WalletsState;
        documents: DocumentsState;
        identity: IdentityState;
        phone: PhoneState;
        history: HistoryState;
        newHistory: NewHistoryState;
        apiKeys: ApiKeysState;
        userActivity: UserActivityState;
        ordersHistory: OrdersHistoryState;
        openOrders: OpenOrdersState;
        sendEmailVerification: EmailVerificationState;
        captchaKeys: GeetestCaptchaState;
        withdrawLimit: WithdrawLimitState;
    };
}

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
});

export function* rootSaga() {
    yield all([
        call(rootApiKeysSaga),
        call(rootAuthSaga),
        call(rootBeneficiariesSaga),
        call(rootConfigsSaga),
        call(rootCurrenciesSaga),
        call(rootEmailVerificationSaga),
        call(rootGeetestCaptchaSaga),
        call(rootHandleAlertSaga),
        call(rootHistorySaga),
        call(rootKlineFetchSaga),
        call(rootLabelSaga),
        call(rootMarketsSaga),
        call(rootMemberLevelsSaga),
        call(rootNewHistorySaga),
        call(rootOpenOrdersSaga),
        call(rootOrderBookSaga),
        call(rootOrdersHistorySaga),
        call(rootOrdersSaga),
        call(rootPasswordSaga),
        call(rootProfileSaga),
        call(rootRecentTradesSaga),
        call(rootSendCodeSaga),
        call(rootSendDocumentsSaga),
        call(rootSendIdentitySaga),
        call(rootUserActivitySaga),
        call(rootWalletsSaga),
        call(rootWithdrawLimitSaga),
    ]);
}
