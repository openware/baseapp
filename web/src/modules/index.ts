import { combineReducers } from 'redux';
import { all, call } from 'redux-saga/effects';
import { adminReducer, publicReducer, userReducer } from './app';
import { ConfigUpdateState, rootConfigUpdateSaga } from './admin/config';
import { AlertState, rootHandleAlertSaga } from './public/alert';
import { BlockchainsState } from './public/blockchains';
import { BlocklistAccessState, rootBlocklistAccessSaga } from './public/blocklistAccess';
import { ConfigsState, rootConfigsSaga } from './public/configs';
import { CurrenciesState } from './public/currencies';
import { ErrorHandlerState, rootErrorHandlerSaga } from './public/errorHandler';
import { GlobalSettingsState } from './public/globalSettings';
import { GridLayoutState } from './public/gridLayout';
import { LanguageState } from './public/i18n';
import { KlineState, rootKlineFetchSaga } from './public/kline';
import { MarketsState, rootMarketsSaga } from './public/markets';
import { MemberLevelsState, rootMemberLevelsSaga } from './public/memberLevels';
import { DepthIncrementState, DepthState, OrderBookState, rootOrderBookSaga } from './public/orderBook';
import { RecentTradesState, rootRecentTradesSaga } from './public/recentTrades';
import { ApiKeysState } from './user/apiKeys';
import { rootApiKeysSaga } from './user/apiKeys/sagas';
import { AuthState, rootAuthSaga } from './user/auth';
import { AbilitiesState, rootAbilitiesSaga } from './user/abilities';
import { BeneficiariesState, rootBeneficiariesSaga } from './user/beneficiaries';
import { GeetestCaptchaState, rootGeetestCaptchaSaga } from './user/captcha';
import { DocumentationState, rootDocumentationSaga } from './user/documentation';
import { EmailVerificationState, rootEmailVerificationSaga } from './user/emailVerification';
import { HistoryState, rootHistorySaga } from './user/history';
import { InternalTransfersState, rootInternalTransfersSaga } from './user/internalTransfers';
import { AddressesState, rootSendAddressesSaga } from './user/kyc/addresses';
import { DocumentsState, rootSendDocumentsSaga } from './user/kyc/documents';
import { IdentityState, rootSendIdentitySaga } from './user/kyc/identity';
import { LabelState, rootLabelSaga } from './user/kyc/label';
import { PhoneState, rootSendCodeSaga } from './user/kyc/phone';
import { OpenOrdersState, rootOpenOrdersSaga } from './user/openOrders';
import { OrdersState, rootOrdersSaga } from './user/orders';
import { OrdersHistoryState, rootOrdersHistorySaga } from './user/ordersHistory';
import { PasswordState, rootPasswordSaga } from './user/password';
import { ProfileState, rootProfileSaga } from './user/profile';
import { rootUserActivitySaga, UserActivityState } from './user/userActivity';
import { rootWalletsSaga, WalletsState } from './user/wallets';
import { QuickExchangeState, rootQuickExchangeSaga } from './user/quickExchange';
import { rootWithdrawLimitSaga, WithdrawLimitState } from './user/withdrawLimit';
import { MarketsAdminState, rootMarketsAdminSaga } from './admin/markets';
import { PlatformCreateState, rootPlatformCreateSaga } from './admin/platform';
import { P2PState, rootP2PSaga } from './public/p2p';
import { PaymentMethodState, rootPaymentMethodSaga } from './user/paymentMethod';
import { P2POffersState, rootP2POffersSaga } from './user/p2pOffers';
import { P2PTransfersState, rootP2PTransfersSaga } from './user/p2pTransfers';
import { P2POrdersState, rootP2POrdersSaga } from './user/p2pOrders';
import { P2PDisputeState, rootP2PDisputeSaga } from './user/p2pDispute';
import { rootFeeGroupSaga, FeeGroupState } from './user/feeGroup';
import { rootWithdrawLimitsSaga, WithdrawLimitsState } from './public/withdrawLimits';

export * from './admin/config';
export * from './admin/markets';
export * from './admin/platform';
export * from './public/alert';
export * from './public/blockchains';
export * from './public/blocklistAccess';
export * from './public/configs';
export * from './public/currencies';
export * from './public/errorHandler';
export * from './public/globalSettings';
export * from './public/i18n';
export * from './public/kline';
export * from './public/markets';
export * from './public/memberLevels';
export * from './public/orderBook';
export * from './public/recentTrades';
export * from './public/p2p';
export * from './public/withdrawLimits';
export * from './user/apiKeys';
export * from './user/auth';
export * from './user/beneficiaries';
export * from './user/captcha';
export * from './user/documentation';
export * from './user/emailVerification';
export * from './user/history';
export * from './user/internalTransfers';
export * from './user/kyc';
export * from './user/openOrders';
export * from './user/orders';
export * from './user/ordersHistory';
export * from './user/password';
export * from './user/profile';
export * from './user/userActivity';
export * from './user/wallets';
export * from './user/feeGroup';
export * from './user/withdrawLimit';
export * from './user/quickExchange';
export * from './user/abilities';
export * from './user/paymentMethod';
export * from './user/p2pOffers';
export * from './user/p2pOrders';
export * from './user/p2pTransfers';
export * from './user/p2pDispute';

export interface RootState {
    public: {
        alerts: AlertState;
        blockchains: BlockchainsState;
        blocklistAccess: BlocklistAccessState;
        globalSettings: GlobalSettingsState;
        configs: ConfigsState;
        currencies: CurrenciesState;
        depth: DepthState;
        errorHandler: ErrorHandlerState;
        i18n: LanguageState;
        incrementDepth: DepthIncrementState;
        kline: KlineState;
        markets: MarketsState;
        memberLevels: MemberLevelsState;
        orderBook: OrderBookState;
        recentTrades: RecentTradesState;
        rgl: GridLayoutState;
        p2p: P2PState;
        withdrawLimits: WithdrawLimitsState,
    };
    user: {
        abilities: AbilitiesState;
        addresses: AddressesState;
        apiKeys: ApiKeysState;
        auth: AuthState;
        beneficiaries: BeneficiariesState;
        captcha: GeetestCaptchaState;
        documentation: DocumentationState;
        documents: DocumentsState;
        history: HistoryState;
        identity: IdentityState;
        internalTransfers: InternalTransfersState;
        label: LabelState;
        openOrders: OpenOrdersState;
        orders: OrdersState;
        ordersHistory: OrdersHistoryState;
        password: PasswordState;
        phone: PhoneState;
        profile: ProfileState;
        sendEmailVerification: EmailVerificationState;
        userActivity: UserActivityState;
        wallets: WalletsState;
        withdrawLimit: WithdrawLimitState;
        feeGroup: FeeGroupState;
        quickExchange: QuickExchangeState;
        paymentMethod: PaymentMethodState;
        p2pOffers: P2POffersState;
        p2pTransfers: P2PTransfersState;
        p2pOrders: P2POrdersState;
        p2pDispute: P2PDisputeState;
    };
    admin: {
        configUpdate: ConfigUpdateState;
        markets: MarketsAdminState;
        platform: PlatformCreateState;
    };
}

export const rootReducer = combineReducers({
    public: publicReducer,
    user: userReducer,
    admin: adminReducer,
});

export function* rootSaga() {
    yield all([
        call(rootWithdrawLimitsSaga),
        call(rootFeeGroupSaga),
        call(rootQuickExchangeSaga),
        call(rootAbilitiesSaga),
        call(rootApiKeysSaga),
        call(rootAuthSaga),
        call(rootBeneficiariesSaga),
        call(rootBlocklistAccessSaga),
        call(rootConfigUpdateSaga),
        call(rootDocumentationSaga),
        call(rootEmailVerificationSaga),
        call(rootErrorHandlerSaga),
        call(rootGeetestCaptchaSaga),
        call(rootHandleAlertSaga),
        call(rootHistorySaga),
        call(rootInternalTransfersSaga),
        call(rootKlineFetchSaga),
        call(rootLabelSaga),
        call(rootMarketsAdminSaga),
        call(rootMarketsSaga),
        call(rootMemberLevelsSaga),
        call(rootOpenOrdersSaga),
        call(rootOrderBookSaga),
        call(rootOrdersHistorySaga),
        call(rootP2PSaga),
        call(rootOrdersSaga),
        call(rootPasswordSaga),
        call(rootPlatformCreateSaga),
        call(rootProfileSaga),
        call(rootRecentTradesSaga),
        call(rootSendAddressesSaga),
        call(rootSendCodeSaga),
        call(rootSendDocumentsSaga),
        call(rootSendIdentitySaga),
        call(rootUserActivitySaga),
        call(rootWalletsSaga),
        call(rootWithdrawLimitSaga),
        call(rootQuickExchangeSaga),
        call(rootPaymentMethodSaga),
        call(rootP2POffersSaga),
        call(rootP2POrdersSaga),
        call(rootP2PTransfersSaga),
        call(rootP2PDisputeSaga),
        call(rootConfigsSaga),
    ]);
}
