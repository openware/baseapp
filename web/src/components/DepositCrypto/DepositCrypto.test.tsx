import { render } from '@testing-library/react';
import React from 'react';
import { TestComponentWrapper } from 'src/lib/test';
import { Wallet } from '../../modules';
import { DepositCrypto } from './';

describe('DepositCrypto', () => {
    let component;
    const handleOnCopy = jest.fn();
    const handleGenerateAddress = jest.fn();
    const wallet: Wallet = {
        currency: 'eth',
        name: '',
        fixed: 0,
        type: 'coin',
        account_type: '',
        networks: [],
        deposit_addresses: [
            {
                address: '0xq',
                currencies: ['eth'],
                blockchain_key: 'erc-20',
            },
        ],
    };

    beforeEach(() => {
        component = render(
            <TestComponentWrapper>
                <DepositCrypto
                    dimensions={118}
                    error={'error123'}
                    handleGenerateAddress={handleGenerateAddress}
                    handleOnCopy={handleOnCopy}
                    text={'text123'}
                    wallet={wallet}
                    disabled={false}
                    network="erc-20"
                />
                ,
            </TestComponentWrapper>,
        );
    });

    it('should contains QRCode', () => {
        expect(component.container.querySelector('.qr-code-wrapper')).toBeInTheDocument();
    });

    it('should contains CopyableTextField', () => {
        expect(component.container.querySelector('.cr-copyable-text-field')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        expect(component.container).toMatchSnapshot();
    });
});
