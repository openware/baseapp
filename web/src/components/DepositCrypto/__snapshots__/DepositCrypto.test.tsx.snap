// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositCrypto should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <DepositCrypto
    dimensions={118}
    disabled={false}
    error="error123"
    handleGenerateAddress={[MockFunction]}
    handleOnCopy={[MockFunction]}
    text="text123"
    wallet={
      Object {
        "currency": "eth",
        "fee": 0,
        "fixed": 0,
        "name": "",
        "type": "coin",
      }
    }
  />
  ,
</ContextProvider>
`;
